### PR TITLE
Upgrade Solidity to v0.8.11

### PR DESCRIPTION
**Pull request by sept-en**
## Dependencies
* Upgrade Solidity to v0.8.11;
* Upgrade OpenZeppelin to v4.4;
* Add Hardhat v2.8;
* Use `rainbow-bridge-sol` gitpkg instead of `rainbow-bridge`.

## Other
* Solidity: flatten contracts using Hardhat. Update dist script.
* Update build files after updating for Hardhat-flattening.
* Replace previous contracts' build artifacts (separated into 2 files: ABI and BIN) with new Hardhat artifacts (one single file for ABI and BIN).

### CODE REVIEW

**Comment by 0x3bfc** [comment]
@sept-en Is that PR ready for review ?

File: erc20-connector/truffle-config.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ module.exports = {
     },
```
Comment (0x3bfc) [inline erc20-connector/truffle-config.js:19]: I am wondering why we have truffle and hardhat configs ?

**Review by 0x3bfc — APPROVED** [review]
LGTM ... with minor comment

**Review by mfornet — APPROVED** [review]
LGTM

**Review by karim-en — APPROVED** [review]

**Pull request merged**